pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -1695,6 +1727,21 @@ cdef class _Timedelta(timedelta):
             self = self._as_creso(other._creso)
         return self, other
 
+    def _get_pytimedelta_days(self):
+        # exposing C-API functions for testing purposes only
+        # consistent behavior is NOT guaranteed
+        return PyDateTime_DELTA_GET_DAYS(self)
+
+    def _get_pytimedelta_seconds(self):
+        # exposing C-API functions for testing purposes only
+        # consistent behavior is NOT guaranteed
+        return PyDateTime_DELTA_GET_SECONDS(self)
+
+    def _get_pytimedelta_microseconds(self):
+        # exposing C-API functions for testing purposes only
+        # consistent behavior is NOT guaranteed
+        return PyDateTime_DELTA_GET_MICROSECONDS(self)
+
```

Comment:
Would appreciate any ideas for alternatives on how to expose these C-API functions for pytest. It doesn't seem that there is a direct way to access cdef functions in Pytest unless they are wrapped in a Python function, and I saw this done similarly in the codebase with how _from_value_and_reso exposes the cdef function _timedelta_from_value_and_reso.

Ultimately, I fear that exposing these here will just make the problem worse, as perhaps someone will find this and start trying to use these wrapped C-APIs directly in their Python code. I hope that the comment makes it clear that this usage won't be supported, though.

Reply:
Instead, could we just use pyarrow to indirectly test that this works? e.g. a unit tests that tests https://github.com/apache/arrow/issues/37291#issue-1860281497

Reply:
Sure, so long as there aren't any objections to importing pyarrow in the tests! I see some other test files do use it, so I'll go ahead and go for this method.

Reply:
Thanks again for the suggestion @mroeschke . After looking at it further, one hesitancy I have with using pyarrow for these test cases is that we expect the pyarrow side to be fixed at some point to use the Pandas interface rather than the C-APIs, per the[ comment](https://github.com/apache/arrow/issues/37291#issuecomment-1687719702) from @jorisvandenbossche in the issue on the pyarrow side.

In other words, in using pyarrow for tests, we'd be testing pyarrow's interface, not the C-APIs. So long as this issue is regarding the C-APIs more generally, I suggest we test these directly (unless we foresee a danger to exposing these functions, in which case I agree pyarrow is a better option).

Reply:
I see. Generally we don't have any testing with Pythons C APIs, so I'm not sure how much pandas can guarantee compatibility. I just suggested the test case in the PyArrow issue to test a pragmatic real world use case